src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+                            me->SetReactState(ReactStates::REACT_PASSIVE);
+                            me->SetFacingToObject(zombie);
+                            me->SetTarget(ObjectGuid::Empty);
+                            state = STATE_GLUTH_EATING;
+
+                            events.ScheduleEvent(EVENT_KILL_ZOMBIE_SINGLE, 2.5 * IN_MILLISECONDS);
+                        }
+
+                        events.ScheduleEvent(EVENT_SEARCH_ZOMBIE_SINGLE, 7 * IN_MILLISECONDS);
                         break;
+                    }
+                    case EVENT_KILL_ZOMBIE_SINGLE:
+                    {
+                        Creature* zombieToBeEaten = ObjectAccessor::GetCreature(*me, zombieToBeEatenGUID);
+                        if (zombieToBeEaten && zombieToBeEaten->IsAlive() && zombieToBeEaten->IsWithinDistInMap(me, 10.0))
+                            DoCast(zombieToBeEaten, SPELL_ZOMBIE_CHOW_SEARCH_SINGLE); // do the killing + healing in done inside by spell script see below
+
+                        zombieToBeEatenGUID = ObjectGuid::Empty;
+                        state = STATE_GLUTH_NORMAL;
+
+                        // and then return on primary target
+                        me->SetReactState(ReactStates::REACT_AGGRESSIVE);
+                        me->SetFacingToObject(nullptr);
+                        me->SetTarget(me->GetVictim()->GetGUID());
+                        me->GetMotionMaster()->MoveChase(me->GetVictim());
+
+                        break;
+                    }
+                    case EVENT_SEARCH_ZOMBIE_MULTI:
+                    {
+                        if (state == STATE_GLUTH_EATING) // skip and simply wait for the next occurence
+                            break;
+
+                        Creature* zombie = nullptr;
+                        for (SummonList::const_iterator itr = summons.begin(); !zombie && itr != summons.end(); ++itr)
+                            if (ObjectAccessor::GetCreature(*me, *itr)->GetExactDist2d(me) <= 18.0)
+                                zombie = ObjectAccessor::GetCreature(*me, *itr);
+                        if (zombie)
+                        {
+                            Talk(EMOTE_DEVOURS_ALL);
+                            DoCast(SPELL_ZOMBIE_CHOW_SEARCH_MULTI);
+                        }
+                        break;
+                    }
+                }
+            }
+
+            if (me->HasReactState(REACT_AGGRESSIVE))
+                DoMeleeAttackIfReady();
+        }
+
+        void DoAction(int32 action) override
+        {
+            switch (action)
+            {
+                case DO_ZOMBIE_RUN: // This action handles the change in the zombies' behavior after a decimate.
+                {
+                    Creature* zombie = nullptr;
+                    for (SummonList::const_iterator itr = summons.begin(); itr != summons.end(); ++itr)
+                    {
+                        zombie = ObjectAccessor::GetCreature(*me, *itr);
```

Comment:
this comes for the summon list. there is no null reference in the summon list. If a zombie dies, line 121 is used to remove it from the list. if the list is empty, itr == summons.end() equals true as soon as the loop starts therefore no problem of null pointer.